src/gmp/models/basemodel.ts

```diff
@@ -3,29 +3,37 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 
+import dayjs, {Dayjs} from 'dayjs';
+import {z} from 'zod';
 import {Date as GmpDate} from 'gmp/models/date';
 import {parseDate} from 'gmp/parser';
 import {isDefined, isString} from 'gmp/utils/identity';
 
 /**
  * Represents the data returned by the backend
  */
-export interface BaseModelElement {
-  _id?: string;
-  creation_time?: string;
-  modification_time?: string;
-  type?: string | number;
-}
+
+export const BaseModelElementSchema = z.object({
+  _id: z.string().optional(),
+  creation_time: z.string().optional(),
+  modification_time: z.string().optional(),
+  type: z.union([z.string(), z.number()]).optional(),
+});
+
+export type BaseModelElement = z.infer<typeof BaseModelElementSchema>;
 
 /**
  * Represents the properties to create a BaseModel instance.
  */
-export interface BaseModelProperties {
-  _type?: string;
-  creationTime?: GmpDate;
-  id?: string;
-  modificationTime?: GmpDate;
-}
+
+export const BaseModelPropertiesSchema = z.object({
+  _type: z.string().optional(),
+  creationTime: z.instanceof(dayjs as unknown as typeof Dayjs).optional(),
```

Comment:
GmpDate should be the type of date
```suggestion
  creationTime: z.instanceof(date as GmpDate).optional(),
```